singer_sdk/sinks/sql.py

```diff
@@ -150,10 +150,18 @@ def conform_name(self, name: str, object_type: Optional[str] = None) -> str:
         Returns:
             The name transformed to snake case.
         """
-        # strip non-alphanumeric characters, keeping - . _ and spaces
+        # strip non-alphanumeric characters
         name = re.sub(r"[^a-zA-Z0-9_\-\.\s]", "", name)
-        # convert to snakecase
-        name = snakecase(name)
+        # strip leading/trailing whitespace,
+        # transform to lowercase and replace - . and spaces to _
+        name = (
+            name.lower()
+            .lstrip()
+            .rstrip()
+            .replace(".", "_")
+            .replace("-", "_")
```

Comment:
This replacement of `-` with `_` deals with (in a great way) some the interop issues we were seeing with tap-dynamodb and tap-mongodb, where tables are often named like `finance-billables-prod` which would cause the target to try to dump everything into a single table called "prod".

👍

Reply:
Given the timeline and priority, I would not block on the `codecoverage` alert. 

For my part, I'd suggest to go ahead and merge when ready. 🚢